tests/otagent/src/test/java/org/hawkular/apm/tests/otagent/server/http/JavaxServletAsyncServerITest.java

```diff
@@ -0,0 +1,296 @@
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        }
+
+        assertEquals(2, getApmMockServer().getTraces().size());
+
+        List<Consumer> consumers = new ArrayList<Consumer>();
+        NodeUtil.findNodes(getApmMockServer().getTraces().get(0).getNodes(), Consumer.class, consumers);
+        NodeUtil.findNodes(getApmMockServer().getTraces().get(1).getNodes(), Consumer.class, consumers);
+
+        assertEquals("Expecting 1 consumers", 1, consumers.size());
+
+        Consumer testConsumer = consumers.get(0);
+
+        assertEquals(path, testConsumer.getUri());
+
+        if (urlstr.endsWith(QUERY_STRING)) {
+            assertEquals(QUERY_STRING,
+                    testConsumer.getProperties(Constants.PROP_HTTP_QUERY).iterator().next().getValue());
+        }
+
+        assertEquals(method, testConsumer.getOperation());
+
+        if (fault) {
+            assertEquals(1, testConsumer.getProperties(Constants.PROP_FAULT).size());
+            assertEquals("Unauthorized", testConsumer.getProperties(Constants.PROP_FAULT).iterator().next().getValue());
+            assertEquals("401",
+                    testConsumer.getProperties(Constants.PROP_FAULT_CODE).iterator().next().getValue());
+        }
+    }
+
+    public static class EmbeddedAsyncServlet extends HttpServlet {
+
+        private static final long serialVersionUID = 1L;
+
+        @Override
+        protected void service(HttpServletRequest request, HttpServletResponse response)
+                throws ServletException, IOException {
+
+            final AsyncContext ctxt = request.startAsync();
+            ctxt.start(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        InputStream is = request.getInputStream();
+
+                        byte[] b = new byte[is.available()];
+                        is.read(b);
+
+                        is.close();
+
+                        System.out.println("REQUEST(ASYNC INPUTSTREAM) RECEIVED: " + new String(b));
+
+                        response.setContentType("text/html; charset=utf-8");
+
+                        if (request.getHeader("test-fault") != null) {
+                            response.sendError(401, "Unauthorized");
+                        } else {
+                            response.setStatus(HttpServletResponse.SC_OK);
+
+                            if (request.getHeader("test-no-data") == null) {
+                                OutputStream os = response.getOutputStream();
+
+                                byte[] resp = HELLO_WORLD_RESPONSE.getBytes();
+
+                                os.write(resp, 0, resp.length);
+
+                                os.flush();
+                                os.close();
+                            }
+                        }
+                    } catch (Exception e) {
+                        fail("Failed: " + e);
```

Comment:
This should not be a test failure, but just report the error as it is in handling the servlet request - have updated.